script.js

```diff
@@ -0,0 +1,364 @@
+initializeCalculator();
+function initializeCalculator() {
+  let first = {value: "", unsaved: true, saved: false, integer: true};
+  let second = {value: "", unsaved: true, saved: false, integer: true};
+  let operator = {value: "", unsaved: true, saved: false};
+  let result = "";
+  let allowNewNumber = false;
+
+  listenClicks();
+  listenKeys();
+
+  function listenKeys() {
+    window.addEventListener("keydown", event => {
+      const key = document.querySelector(`[data-key="${event.key}"]`);
+      if (key) {
+        bindEvent(key.id);
+        key.focus(); // Safari and Firefox don't handle focus in CSS files
+      }
+    });
+  }
+
+  function listenClicks() {
+    const buttons = document.querySelectorAll(".btn");
+    buttons.forEach(button => {
+      button.addEventListener("click", event => {
+        const click = event.target;
+        bindEvent(click.id);
+        click.focus(); // Safari and Firefox don't handle focus in CSS files
+      });
+    });
+  }
+
+  function bindEvent(event) {
+    if (["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].includes(event)) {
+      addOperand(event);
+    }
+    if (["+", "-", "÷", "×", "^"].includes(event)) {
+      addOperator(event);
+    }
+    switch (event) {
+      case "=":
+        evaluateExpression(first, operator, second);
+        break;
+      case ".":
+        addDecimal(event);
+        break;
+      case "+/-":
+        getOpposite();
+        break;
+      case "%":
+        getPercentage();
+        break;
+      case "ac":
+        clearExpression();
+        break;
+    }
+    updateClearbutton("AC", "C");
+    updateDisplay(first.value, operator.value, second.value, result);
+    console.table({
+      first: first,
+      operator: operator,
+      second: second,
+      result: result,
+      new: allowNewNumber
+    });
+  }
+
+  function addOperand(operandEvent) {
```

Comment:
`operandEvent` is not a very good name, since the parameter is not an event but `operand`